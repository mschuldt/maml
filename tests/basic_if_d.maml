arduino = Arduino(desktop=True)

@block(once)
def test():
    if 1: print(111)
    else: print(222)
        
    if 0: print(111)
    else: print(222)

    false = 0
    true = 1
    
    if true: print(true) 
    else: print(false)
    
    if false: print(true)
    else: print(false)
    
    if true:
        print(3)
    else:
        print(5)

    if true:
        print(300)
        
    if false:
        print(300)
        
    if false:
        print(500)
    elif true:
        print(100)

    if false:
        pass
    elif false:
        pass
    elif false:
        pass
    elif true:
        print(888)
    else:
        print(11111)
        
    if 1+1+1:
        print(91)
        
    if test_add(1, 2):
        print(92)

    if test_print(0):
        print(93)

    if test_print(11):
        print(94)

    if test_print(test_print(12)+1):
        print(95)
        
    die(1)

arduino.send(test)
