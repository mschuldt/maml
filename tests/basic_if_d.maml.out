Initializing avm...
Ready.

111
222
1
0
3
300
100
888
91
92
-> 0
-> 11
94
-> 12
-> 13
95
bye.
